fastembed/common/model_management.py

```diff
@@ -107,36 +110,89 @@ def download_files_from_huggingface(
         Args:
             hf_source_repo (str): Name of the model on HuggingFace Hub, e.g. "qdrant/all-MiniLM-L6-v2-onnx".
             cache_dir (Optional[str]): The path to the cache directory.
-            extra_patterns (Optional[list[str]]): extra patterns to allow in the snapshot download, typically
+            extra_patterns (list[str]): extra patterns to allow in the snapshot download, typically
                 includes the required model files.
             local_files_only (bool, optional): Whether to only use local files. Defaults to False.
         Returns:
             Path: The path to the model directory.
         """
+        verified_metadata = False
+
+        def _verify_files_from_metadata(
+            model_dir: Path, stored_metadata: dict[str, Any], hf_source_repo: str
+        ) -> bool:
+            for rel_path, meta in stored_metadata.items():
+                file_path = model_dir / rel_path
+                file_name = file_path.name
+
+                try:  # if there's network, check with hf
+                    url = hf_hub_url(hf_source_repo, file_name)
+                    hf_metadata = get_hf_file_metadata(url)
+                    if (
+                        not file_path.exists()
+                        or hf_metadata.size != meta["size"]
+                        or hf_metadata.commit_hash != meta["commit_hash"]
+                    ):
+                        return False
+                except Exception:  # if not, only check local size
+                    if not file_path.exists() or file_path.stat().st_size != meta["size"]:
+                        return False
+            return True
+
+        def _save_file_metadata(model_dir: Path, hf_source_repo: str) -> None:
+            metadata = {}
+            for file_path in model_dir.rglob("*"):
+                if file_path.is_file() and file_path.name != cls.METADATA_FILE:
+                    try:  # there's network at least first time
+                        url = hf_hub_url(hf_source_repo, file_path.name)
+                        hf_metadata = get_hf_file_metadata(url)
+                        metadata[str(file_path.relative_to(model_dir))] = {
+                            "size": hf_metadata.size,
+                            "commit_hash": hf_metadata.commit_hash,
+                        }
```

Comment:
can we retrieve information about the whole repository instead of retrieving files one by one?